src/RollOfHonour.Web/Pages/Search/Results.razor

```diff
@@ -292,12 +303,52 @@
         }
     }
 
+    private async Task QueryPeopleByRegiment(bool stateChangeNeeded, int pageIndex)
+    {
+        PersonSearchResults = new();
+        if (_selectedQueryType == QueryType.Regiment)
+        {
+            var query = BuildRegimentPersonQuery();
+            var filters = new Filters(_selectedWar, _dateOfDeathFilter, _dateOfBirthFilter, _selectedPersonType, _regimentIds);
+            Result<PaginatedList<Person>> results = await _searchService.PersonSearchByRegimentName(query, filters, pageIndex, _pageSize);
+
+            if (results.IsSuccess is not true)
+            {
+                return;
+            }
+
+            if (!results.Value.Any())
+            {
+                return;
+            }
+
+
+            PersonSearchResults = results.Value;
+            if (stateChangeNeeded)
+            {
+                StateHasChanged();
+            }
+        }
+    }
+
     private async Task FilterSearch()
     {
         _isLoading = true;
         PageIndex = 1;
         _ = await JSRuntime.InvokeAsync<bool>("scrollToElementId", "results-jump-to");
-        await QueryPeople(true, PageIndex);
+        switch (_selectedQueryType)
+        {
+            case QueryType.Regiment:
+                await QueryPeopleByRegiment(true, PageIndex);
+                break;
+            case QueryType.Person:
+                await QueryPeople(true, PageIndex);
+                break;
+            case QueryType.Memorial:
+                break;
+            default:
+                throw new ArgumentOutOfRangeException();
```

Comment:
Same as the other point

Reply:
Good spot, arguably it's redundant code, so I've removed it.